src/main.rs

```diff
@@ -55,35 +57,43 @@ impl fmt::Display for Info {
             None => Color::White,
         };
 
-        writeln!(
-            buffer,
-            "{}{}",
-            "Project: ".color(color).bold(),
-            self.project_name
-        )?;
+        if !self.disable_fields.contains(&InfoFields::Project) {
+            writeln!(
+                buffer,
+                "{}{}",
+                "Project: ".color(color).bold(),
+                self.project_name
+            )?;
+        }
 
-        writeln!(
-            buffer,
-            "{}{}",
-            "HEAD: ".color(color).bold(),
-            self.current_commit
-        )?;
+        if !self.disable_fields.contains(&InfoFields::HEAD) {
+            writeln!(
+                buffer,
+                "{}{}",
+                "HEAD: ".color(color).bold(),
+                self.current_commit
+            )?;
+        }
```

Comment:
Running `Vec::contains` multiple times is kind of computationally intensive, as each call will iterate through until finding a match. See source code [here](https://doc.rust-lang.org/src/core/slice/mod.rs.html#1282-1286) and [here](https://doc.rust-lang.org/src/core/slice/mod.rs.html#5557-5561).

Iterating through once and keeping track `bool` values might be better.
Example:
```rust
let mut no_project = false;
// etc.
for field in self.disable_fields.iter() {
    match field {
        InfoFields::Project => no_project = true,
        // etc.
    }
}

if !no_project {
    // write project
}
```

Reply:
I decided to create a struct of bools just to make it easier to track.